tools/cldr-code/src/test/java/org/unicode/cldr/util/TestCLDRPaths.java

```diff
@@ -0,0 +1,50 @@
+package org.unicode.cldr.util;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import java.io.File;
+import org.junit.jupiter.api.Test;
+
+public class TestCLDRPaths {
+
+    @Test
+    void TestCanUseArchiveDirectory() {
+        if (!canUseArchiveDirectory()) {
+            // We print this warning as part of the unit tests.
+            System.err.println(
+                    "WARNING: skipping using cldr-archive. Ideally, -DNO_CLDR_ARCHIVE=false and see <https://cldr.unicode.org/development/creating-the-archive>");
+        }
+    }
+
+    /**
+     * @return true if it's OK to read CLDRPaths.ARCHIVE_DIRECTORY, false to skip.
+     */
+    public static final boolean canUseArchiveDirectory() {
+        if (CLDRConfig.getInstance().getProperty("NO_CLDR_ARCHIVE", false)) {
+            return false; // skip, NO_CLDR_ARCHIVE is set.
+        }
+
+        final File archiveDir = new File(CLDRPaths.ARCHIVE_DIRECTORY);
+        if (!archiveDir.isDirectory()) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "Could not read archive directory %s. "
+                                    + "Please: "
+                                    + "1) setup the archive, <https://cldr.unicode.org/development/creating-the-archive>, "
```

Comment:
Ambiguous: does this mean #1 and (#2 or #3), or (#1 and #2) or #3?

Reply:
i've changed this (see example run below).

I took out the option of changing `-DARCHIVE` by instead folding that into the informative. message (`-DARCHIVE=/a/b/c`) - so if /a/b/c is the wrong path the user can correct that property.  

Will need to update the docs page after this of course.